src/app/utils/dataLoaderUtil.js

```diff
@@ -3,122 +3,61 @@ import {
   },
   search: {
-    apiRoute: (matchData, route) => `${route}?${matchData[1]}`,
     action: updateSearchResultsPage,
-    errorMessage: 'Error attempting to make an ajax request to search',
+    path: 'search',
+    params: '',
   },
   holdRequest: {
-    apiRoute: (matchData, route) => route.replace(':bibId-:itemId', matchData[1]),
-    serverParams: (matchData, req) => {
-      const params = matchData[1].match(/\w+/g);
-      if (params[0]) req.params.bibId = params[0];
-      if (params[1]) req.params.itemId = params[1];
-    },
     action: updateHoldRequestPage,
-    errorMessage: 'Error attempting to make ajax request for hold request',
+    path: 'hold/request',
+    params: '/:bibId-:itemId',
   },
-});
-
-const matchingPathData = (location) => {
-  const {
-    pathname,
-    search,
-  } = location;
+};
 
-  return pathInstructions
-    .map(instruction => ({
-      matchData: (pathname + search).match(instruction.expression),
-      pathType: instruction.pathType,
-    }))
-    .find(pair => pair.matchData)
-    || { matchData: null, pathType: null };
+const successCb = (pathType, dispatch) => (response) => {
+  dispatch(routes[pathType].action(response.data));
 };
 
-function loadDataForRoutes(location, req, routeMethods, realRes) {
-  const routes = routesGenerator(location);
-  const {
-    matchData,
-    pathType,
-  } = matchingPathData(location);
+function loadDataForRoutes(location, dispatch) {
+  const { pathname } = location;
+
+  const matchingPath = Object.entries(routes).find(([pathKey, pathValue]) => {
+    const { path } = pathValue;
+    return pathname.match(`${baseUrl}/${path}`);
+  });
+
+  if (!matchingPath) return null;
+
+  const pathType = matchingPath[0];
+
+  const errorCb = (error) => {
+    console.error(
+      `Error attempting to make ajax request for ${pathType}`,
+      error,
+    );
+  };
 
-  if (routes[pathType]) {
-    const {
-      apiRoute,
-      action,
-      errorMessage,
-      serverParams,
-    } = routes[pathType];
-    const route = routePaths[pathType];
-    const successCb = (response) => {
-      dispatch(action(response.data, location));
-    };
-    const errorCb = (error) => {
-      console.error(
-        errorMessage,
-        error,
-      );
-    };
-    if (req) {
-      console.log('making server side call');
-      if (serverParams) serverParams(matchData, req);
-      return new Promise((resolve) => {
-        const res = {
-          redirect: url => realRes.redirect(url),
-          json: (data) => {
-            resolve({ data });
-          },
-        };
-        return routeMethods[pathType](req, res);
-      })
-        .then(({ data }) => {
-          realRes.data = { ...realRes.data, ...data };
-          return realRes;
-        })
-        .catch(errorCb);
-    }
-    console.log('making ajaxCall');
-    return ajaxCall(apiRoute(matchData, route), successCb, errorCb);
-  }
-  return new Promise(resolve => resolve());
+  return ajaxCall(
```

Comment:
This is now for server side as well, right?

Reply:
Server side doesn't really make any API calls directly (and the API calls it makes indirectly already have their own error handling), so I don't think this is needed server side.

Reply:
Realized what I wrote was a little cryptic. On the server side, `dataLoaderUtil` now never gets called. All that happens is that the express router recognizes the path (because of the configuration in `ApiRoutes`), passes the params to the corresponding function, and then passes the result directly into `successCb`, which puts them in the store.